cmd/operator/main.go

```diff
@@ -47,17 +47,19 @@ import (
 var (
 	appVersion = "0.0.12"
 
-	printVersion bool
-	baseImage    string
-	kubeCfgFile  string
-	masterHost   string
+	printVersion           bool
+	baseImage              string
+	kubeCfgFile            string
+	masterHost             string
+	initDaemonsetNamespace string
 )
 
 func init() {
 	flag.BoolVar(&printVersion, "version", false, "Show version and quit")
 	flag.StringVar(&baseImage, "baseImage", "upmcenterprises/docker-elasticsearch-kubernetes:6.1.3_0", "Base image to use when spinning up the elasticsearch components.")
 	flag.StringVar(&kubeCfgFile, "kubecfg-file", "", "Location of kubecfg file for access to kubernetes master service; --kube_master_url overrides the URL part of this; if neither this nor --kube_master_url are provided, defaults to service account tokens")
 	flag.StringVar(&masterHost, "masterhost", "http://127.0.0.1:8001", "Full url to k8s api server")
+	flag.StringVar(&initDaemonsetNamespace, "initDaemonsetNamespace", "default", "Namespace to deploy the sysctl init daemonset into")
```

Comment:
Should we default the namespace to use the one the operator is deployed into?

Reply:
yes to me defaulting to where the operator is deployed makes sense, but you can still customize where it could be deployed. Would you mind updating?

Reply:
Sure thing! From what I can gather there are two ways that a pod can determine which namespace it is running in. The first is with the [Downward API](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#the-downward-api) which allows one to set environment variables which contain metadata about a pod. In fact, the current example for the elasticsearch-operator uses the Downward API to [expose the namespace that the operator is running in already](https://github.com/upmc-enterprises/elasticsearch-operator/blob/master/example/controller.yaml#L67). 

The second approach would be to read the namespace from `/var/run/secrets/kubernetes.io/serviceaccount/namespace`. This file [contains the namespace for the service account the operator is associated](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod). Strictly speaking, this is the namespace of the service account and not the pod, but as far as I can tell the service account for a pod must be in the same namespace as the pod as [the spec for a pod only contains a field for the name of the service account and not its namespace](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2765).

Any preference on which approach to use? The first approach requires that the namespace is explicitly set as an environment variable so it requires slightly more configuration than the second approach, but has the advantage of being explicit.

Reply:
Hey @jeromefroe I think using the downward api like we are today is a good safe option.